### PR TITLE
test: hardcode clippy lints

### PR DESCRIPTION
**Pull request by Manuthor**
add by default, on `ckms` and `cosmian_kms_server` this clippy config:
```
#![deny(
    nonstandard_style,
    refining_impl_trait,
    future_incompatible,
    keyword_idents,
  
  let_underscore,
    rust_2024_compatibility,
    clippy::all,
    clippy::suspicious,
    clippy::complexity,
    clippy::perf,
    clippy::style,
    clippy::pedantic,
    clippy::cargo
)]
```

Also replace in `ckms` all occurences of `Result<Something, CliError>` by `CliResult<Something>`

### CODE REVIEW

File: crate/cli/src/actions/shared/import_key.rs [hunk 1]
```diff
@@ -15,7 +15,10 @@ use cosmian_kms_client::{
 use zeroize::Zeroizing;
 
 use super::utils::{build_usage_mask_from_key_usage, KeyUsage};
-use crate::{actions::console, error::CliError};
+use crate::{
+    actions::console,
+    error::{result::CliResult, CliError},
```
Comment (ccorsin) [inline crate/cli/src/actions/shared/import_key.rs:20]: same here

File: crate/cli/src/actions/google/keypairs/insert_keypairs.rs [hunk 2]
```diff
@@ -9,7 +9,10 @@ use clap::Parser;
 use serde::{Deserialize, Serialize};
 
 use super::KEYPAIRS_ENDPOINT;
-use crate::{actions::google::gmail_client::GmailClient, error::CliError};
+use crate::{
+    actions::google::gmail_client::GmailClient,
+    error::{result::CliResult, CliError},
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/cli/src/actions/google/keypairs/insert_keypairs.rs:14]: indeed. Done in https://github.com/Cosmian/kms/commit/177df68c37757fcbabe92f97d8ceac02e7160299.

File: crate/cli/src/actions/login.rs [hunk 3]
```diff
@@ -25,7 +25,10 @@ use oauth2::{
 use serde::Deserialize;
 use url::Url;
 
-use crate::{cli_bail, error::CliError};
+use crate::{
+    cli_bail,
+    error::{result::CliResult, CliError},
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/cli/src/actions/login.rs:30]: indeed. Done in https://github.com/Cosmian/kms/commit/177df68c37757fcbabe92f97d8ceac02e7160299.

File: crate/cli/src/actions/shared/import_key.rs (hunk 1 above)
Comment (Manuthor) [reply crate/cli/src/actions/shared/import_key.rs:20]: indeed. Done in https://github.com/Cosmian/kms/commit/177df68c37757fcbabe92f97d8ceac02e7160299.

**Pull request merged**